### ISSUE TITLE
add BEFT as a new SFT/PEFT method

### ISSUE THREAD

**Issue by whubaichuan** [google/tunix#1490]
**Is your feature request related to a problem? Please describe.**
I initially want to add our recently published **BEFT** [(paper)](https://arxiv.org/pdf/2509.15974v2) (ACL2026 Main Conference) as **a `sft/peft` method in Tunix**. **However**, I found that the current tunix-LoRA is provided from [qwix](https://github.com/google/qwix), as shown [here](https://github.com/google/tunix/blob/main/examples/qlora_gemma.ipynb).   Moreover, in `peft_trainer` ([link](https://github.com/google/tunix/blob/e0b05530b834aa971ad9c7a15beeda1dc3de422e/tunix/sft/peft_trainer.py#L211)), the lora option is provided from [flax](https://github.com/google/tunix/blob/e0b05530b834aa971ad9c7a15beeda1dc3de422e/tunix/sft/utils.py#L74).

Therefore, I am wondering that where is the best place for contributing other `sft/peft ` methods? 

**Describe the solution you'd like**
Alternatively, will Tunix provide the direct interface for intergrating more `sft/peft` methods?

**Additional context**
I wish get some guidance. Thank you in advance @tianshub @s-noghabi

**Checklist**

- [x] I have searched the existing issues for similar feature requests.
- [x] This is not a support question (please use the "bug template" for that).

**Comment by tianshub** [comment]
@whubaichuan can you outline an implementation plan for BEFT? like what are the proposed code change.

then we can see if it's better to land it in qwix or tunix. Thanks!

**Comment by whubaichuan** [comment]
@tianshub Thanks for your message.

The outline of my proposed code change is **only setting certain specific parameters in LLM as trainable**, while **BEFT is only setting the bias terms of values projection to be trainable** (_if no bias terms in some LLM, BEFT adds a trainable vector as the role of bias terms_). The benefit is the extreme parameter efficiency with a competitive accuracy, especially on few training data, as valided [here](https://github.com/huggingface/peft/pull/3195#pullrequestreview-4204340713).

BTW, if you decide to land it in tunix, I would suggest that Tunix could provide the interface for intergrating more [sft/peft](https://github.com/huggingface/peft/tree/main/src/peft/tuners) methods, such as [prompt-tuning](https://aclanthology.org/2022.acl-short.8/), [prefix tuning](https://aclanthology.org/2021.acl-long.353/), [PiSSA](https://proceedings.neurips.cc/paper_files/paper/2024/hash/db36f4d603cc9e3a2a5e10b93e6428f2-Abstract-Conference.html), etc.

**Comment by tianshub** [comment]
@jshin1394 can you take a look and see if this is a better fit for qwix to support?

**Comment by jshin1394** [comment]
Hi @tianshub, Thank you so much for reaching out to me with this. I totally agree that the BEFT algorithm is a great method that can be implemented using the Qwix provider pattern as it would be similar to the existing Qwix QLoRA implementation in that both selectively update a small subset of parameters.

However, the reason why the QLoRA exists as a core algorithm within Qwix is because of the Q part, the quantized nature of the method. BEFT, on the other hand, as described in the paper, focuses on which existing parameters (the biases) to fine-tune. It doesn't inherently involve new quantization techniques for weights or activations.

Given this, I believe the most suitable place for implementing BEFT would be within the Tunix codebase, as Tunix is designed for post-training and fine-tuning workflows. Tunix could extend an existing Qwix provider and implement BEFT. Qwix will be happy to provide necessary guidance, review and communication during the process! :)
